content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,311 @@
+    - obtains from decrypting the received message a public key `sB`. If `sB` is not a valid public key, the protocol is aborted.
+    - performs `DH(eA,sB)` (which updates a symmetric encryption key);
+    - decrypts the payload to obtain the randomness `r`. 
+    - Computes `H(sB||r)` and checks if this value corresponds to the commitment obtained in step 2. If not, the protocol is aborted.
+    - executes the third handshake message, i.e.
+        - processes and sends his (encrypted) device static key `sA` over `contentTopic`;
+        - performs `DH(sA,eB)` (which updates the symmetric encryption key);
+        - performs `DH(sA,sB)` (which updates the symmetric encryption key);
+        - attaches as payload the (encrypted) commitment randomness `s` used to compute `H(sA||s)`.
+    - Calls Split() and obtains two cipher states to encrypt inbound and outbound messages.
+    
+7. The device `B`:
+
+    - listens to messages sent to `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto` and locally filters only those with Waku payload starting with `messageNametag`. If any, continues.
+    - obtains from decrypting the received message a public key `sA`. If `sA` is not a valid public key, the protocol is aborted.
+    - performs `DH(sA,eB)` (which updates a symmetric encryption key);
+    - performs `DH(sA,sB)` (which updates a symmetric encryption key);
+    - decrypts the payload to obtain the randomness `s`. 
+    - Computes `H(sA||s)` and checks if this value corresponds to the commitment obtained in step 6. If not, the protocol is aborted.
+    - Calls Split() and obtains two cipher states to encrypt inbound and outbound messages.
+
+### The `WakuPairing` for Devices without a Camera
+
+In the above pairing handshake, the QR is by default exposed by device `B` and not by `A` because device `B` locally stores no relevant cryptographic material, 
+so an active local attacker that scans the QR code first 
+would only be able to transfer *his own* session information 
+and get nothing from `A`. 
+
+However, since the user confirms at the end of message `1` that the authorization code is the same on both devices, 
+the role of handhsake initiator and responder can be safely swapped in message `0` and `1`. 
+
+This allows pairing in case device `A` does not have a camera to scan a QR (e.g. a desktop client) while device `B` has.
+
+The resulting handshake would then be:
+```
+WakuPairing2:
+0.   -> eA              {H(sB||r), contentTopicParams, messageNametag}
+     ...
+1.   <- eB, eAeB        {H(sB||r)}   [authcode]
+2.   <- sB, eAsB        {r}
+3.   -> sA, sAeB, sAsB  {s}
+
+{}: payload,    []: user interaction
+```
+
+# Secure Transfer
+
+The pairing phase is designed to be application-agnostic 
+and should be flexible enough to mutually authenticate 
+and allow exchange of cryptographic key material 
+between two devices over a distributed network of Waku2 nodes.
```

Comment:
Should we broaden the scope of this RFC to enable communication between any device then? Current scope is "same user devices).